File: flutter_zoom_videosdk-2.4.0/android/src/main/java/com/flutterzoom/videosdk/FlutterZoomVideoSdkLiveTranscriptionHelper.java
package com.flutterzoom.videosdk;

import android.app.Activity;

import androidx.annotation.NonNull;

import com.flutterzoom.videosdk.convert.FlutterZoomVideoSdkErrors;
import com.flutterzoom.videosdk.convert.FlutterZoomVideoSdkLiveTranscriptionStatus;

import java.util.List;
import java.util.Map;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import us.zoom.sdk.ZoomVideoSDK;
import us.zoom.sdk.ZoomVideoSDKErrors;
import us.zoom.sdk.ZoomVideoSDKLiveTranscriptionHelper;

public class FlutterZoomVideoSdkLiveTranscriptionHelper {

    private Activity activity;

    FlutterZoomVideoSdkLiveTranscriptionHelper(Activity activity) {
        this.activity = activity;
    }

    private ZoomVideoSDKLiveTranscriptionHelper getLiveTranscriptionHelper() {
        ZoomVideoSDKLiveTranscriptionHelper liveTranscriptionHelper = null;
        try {
            liveTranscriptionHelper = ZoomVideoSDK.getInstance().getLiveTranscriptionHelper();
            if (liveTranscriptionHelper == null) {
                throw new Exception("No Live Transcription Helper Found");
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return liveTranscriptionHelper;
    }

    public void canStartLiveTranscription(@NonNull MethodChannel.Result result) {
        result.success(getLiveTranscriptionHelper().canStartLiveTranscription());
    }

    public void getLiveTranscriptionStatus(@NonNull MethodChannel.Result result) {
        result.success(FlutterZoomVideoSdkLiveTranscriptionStatus.valueOf(getLiveTranscriptionHelper().getLiveTranscriptionStatus()));
    }

    public void startLiveTranscription(@NonNull MethodChannel.Result result) {
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getLiveTranscriptionHelper().startLiveTranscription()));
            }
        });
    }

    public void stopLiveTranscription(@NonNull MethodChannel.Result result) {
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getLiveTranscriptionHelper().stopLiveTranscription()));
            }
        });
    }

    public void getAvailableSpokenLanguages(@NonNull MethodChannel.Result result) {
        List<ZoomVideoSDKLiveTranscriptionHelper.ILiveTranscriptionLanguage> languageList = getLiveTranscriptionHelper().getAvailableSpokenLanguages();

        if (languageList == null) {
            result.error("RNZoomVideoSdkLiveTranscriptionHelper::getAvailableSpokenLanguages", "there is no available language for live transcription", null);
            return;
        }

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkILiveTranscriptionLanguage.jsonLanguageArray(languageList));
            }
        });
    }

    public void setSpokenLanguage(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> args = call.arguments();
        int languageID = (Integer) args.get("languageID");

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getLiveTranscriptionHelper().setSpokenLanguage(languageID)));
            }
        });

    }

    public void getSpokenLanguage(@NonNull MethodChannel.Result result) {
        ZoomVideoSDKLiveTranscriptionHelper.ILiveTranscriptionLanguage language = getLiveTranscriptionHelper().getSpokenLanguage();

        if (language == null) {
            result.error("RNZoomVideoSdkLiveTranscriptionHelper::getSpokenLanguage ", "spoken language doesn't exist", null);
            return;
        }

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkILiveTranscriptionLanguage.jsonLanguage(language));
            }
        });

    }

    public void getAvailableTranslationLanguages(@NonNull MethodChannel.Result result) {
        List<ZoomVideoSDKLiveTranscriptionHelper.ILiveTranscriptionLanguage> languageList = getLiveTranscriptionHelper().getAvailableTranslationLanguages();

        if (languageList == null) {
            result.error("RNZoomVideoSdkLiveTranscriptionHelper::getAvailableTranslationLanguages", "there is no available translation language for live transcription", null);
            return;
        }

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkILiveTranscriptionLanguage.jsonLanguageArray(languageList));
            }
        });

    }

    public void setTranslationLanguage(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> args = call.arguments();
        int languageID = (Integer) args.get("languageID");

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getLiveTranscriptionHelper().setTranslationLanguage(languageID)));
            }
        });

    }

    public void getTranslationLanguage(@NonNull MethodChannel.Result result) {
        ZoomVideoSDKLiveTranscriptionHelper.ILiveTranscriptionLanguage language = getLiveTranscriptionHelper().getTranslationLanguage();

        if (language == null) {
            result.error("RNZoomVideoSdkLiveTranscriptionHelper::getTranslationLanguage ", "translation language doesn't exist", null);
            return;
        }

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkILiveTranscriptionLanguage.jsonLanguage(language));
            }
        });
    }

    public void isReceiveSpokenLanguageContentEnabled(@NonNull MethodChannel.Result result) {
        result.success(getLiveTranscriptionHelper().isReceiveSpokenLanguageContentEnabled());
    }

    public void enableReceiveSpokenLanguageContent(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> args = call.arguments();
        boolean enable = (boolean) args.get("enable");

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getLiveTranscriptionHelper().enableReceiveSpokenLanguageContent(enable)));
            }
        });
    }

    public void isAllowViewHistoryTranslationMessageEnabled(@NonNull MethodChannel.Result result) {
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(getLiveTranscriptionHelper().isAllowViewHistoryTranslationMessageEnabled());
            }
        });
    }

    public void getHistoryTranslationMessageList(@NonNull MethodChannel.Result result) {
        List<ZoomVideoSDKLiveTranscriptionHelper.ILiveTranscriptionMessageInfo> messageInfoList = getLiveTranscriptionHelper().getHistoryTranslationMessageList();

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkILiveTranscriptionMessageInfo.jsonMessageInfoArray(messageInfoList));
            }
        });

    }
}

File: flutter_zoom_videosdk-2.4.0/ios/Classes/FlutterZoomVideoSdkLiveTranscriptionHelper.m
#import "FlutterZoomVideoSdkLiveTranscriptionHelper.h"
#import "FlutterZoomVideoSdkLiveTranscriptionLanguage.h"
#import "FlutterZoomVideoSdkILiveTranscriptionMessageInfo.h"
#import "JSONConvert.h"

@implementation FlutterZoomVideoSdkLiveTranscriptionHelper

- (ZoomVideoSDKLiveTranscriptionHelper *)getLiveTranscriptionHelper {
    ZoomVideoSDKLiveTranscriptionHelper* liveTranscriptionHelper = nil;
    @try {
        liveTranscriptionHelper = [[ZoomVideoSDK shareInstance] getLiveTranscriptionHelper];
        if (liveTranscriptionHelper == nil) {
            NSException *e = [NSException exceptionWithName:@"NoRecordingHelperFound" reason:@"No Live Transcription Helper Found" userInfo:nil];
            @throw e;
        }
    } @catch(NSException *e) {
        NSLog(@"%@ - %@", e.name, e.reason);
    }
    return liveTranscriptionHelper;
}

-(void) canStartLiveTranscription: (FlutterResult) result {
    if ([[self getLiveTranscriptionHelper] canStartLiveTranscription]) {
        result(@YES);
    } else {
        result(@NO);
    }
}

-(void) getLiveTranscriptionStatus: (FlutterResult) result {
    result([[JSONConvert ZoomVideoSDKLiveTranscriptionStatusValuesReversed] objectForKey: @([[self getLiveTranscriptionHelper] getLiveTranscriptionStatus])]);
}

-(void) startLiveTranscription: (FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getLiveTranscriptionHelper] startLiveTranscription])]);
   });
}

-(void) stopLiveTranscription: (FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getLiveTranscriptionHelper] stopLiveTranscription])]);
   });
}

-(void) getAvailableSpokenLanguages: (FlutterResult) result {
    NSArray <ZoomVideoSDKLiveTranscriptionLanguage*>* languageList = [[self getLiveTranscriptionHelper] getAvailableSpokenLanguages];

    if (languageList == nil | [languageList count] == 0) {
        result(@"getAvailableSpokenLanguages: No Languages Found");
    }

    result([FlutterZoomVideoSdkLiveTranscriptionLanguage mapLanguageArray:languageList]);
}

-(void) setSpokenLanguage: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getLiveTranscriptionHelper] setSpokenLanguage: [call.arguments[@"languageID"] integerValue]])]);
}

-(void) getSpokenLanguage: (FlutterResult) result {
    ZoomVideoSDKLiveTranscriptionLanguage* language = [[self getLiveTranscriptionHelper] getSpokenLanguage];

    if (language == nil) {
        result(@"getSpokenLanguage: No Languages Found");
    }

    result([FlutterZoomVideoSdkLiveTranscriptionLanguage mapLanguage:language]);
}

-(void) getAvailableTranslationLanguages: (FlutterResult) result {
    NSArray <ZoomVideoSDKLiveTranscriptionLanguage*>* languageList = [[self getLiveTranscriptionHelper] getAvailableTranslationLanguages];

    if (languageList == nil | [languageList count] == 0) {
        result(@"getAvailableTranslationLanguages: No Languages Found");
    }

    result([FlutterZoomVideoSdkLiveTranscriptionLanguage mapLanguageArray:languageList]);
}

-(void) setTranslationLanguage: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getLiveTranscriptionHelper] setTranslationLanguage: [call.arguments[@"languageID"] integerValue]])]);
}

-(void) getTranslationLanguage: (FlutterResult) result {
    ZoomVideoSDKLiveTranscriptionLanguage* language = [[self getLiveTranscriptionHelper] getTranslationLanguage];

    if (language == nil) {
        result(@"getTranslationLanguage: No Languages Found");
    }

    result([FlutterZoomVideoSdkLiveTranscriptionLanguage mapLanguage:language]);
}

-(void) enableReceiveSpokenLanguageContent: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getLiveTranscriptionHelper] enableReceiveSpokenLanguageContent: [call.arguments[@"enabled"] boolValue]])]);
    });
}

-(void) isReceiveSpokenLanguageContentEnabled: (FlutterResult) result {
    result(@([[self getLiveTranscriptionHelper] isReceiveSpokenLanguageContentEnabled]));
}

-(void) isAllowViewHistoryTranslationMessageEnabled: (FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        result(@([[self getLiveTranscriptionHelper] isAllowViewFullTranscriptEnable]));
    });
}

-(void) getHistoryTranslationMessageList: (FlutterResult) result {

    dispatch_async(dispatch_get_main_queue(), ^{
        NSArray <ZoomVideoSDKLiveTranscriptionMessageInfo*>* messageInfoList = [[self getLiveTranscriptionHelper] getHistoryTranslationMessageList];
        result([FlutterZoomVideoSdkILiveTranscriptionMessageInfo mapMessageInfoArray: messageInfoList]);
    });
}

@end
